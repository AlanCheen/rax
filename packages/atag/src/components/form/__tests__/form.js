describe('form', () => {
  const pages = [];

  async function createPage() {
    const page = await global.__BROWSER__.newPage();
    await page.setViewport(global.__VIEWPORT__);
    await page.goto('http://localhost:9002/components/form/__tests__/form.html');
    pages.push(page);
    return page;
  }

  afterAll(async() => {
    for (let i = 0, l = pages.length; i < l; i++) {
      await pages[i].close();
    }
  });

  it('submit', async() => {
    const page = await createPage();
    await page.click('#radioTarget');
    await page.click('#checkboxTarget');
    await page.click('#input');
    await page.keyboard.sendCharacter('hello world');
    await page.click('#submit');
    const submitEventDetail = await page.evaluate('window.__submit_event_detail__');
    expect(submitEventDetail).toMatchSnapshot();
  });

  it('reset', async() => {
    const page = await createPage();
    await page.click('#radioTarget');
    await page.click('#checkboxTarget');
    await page.click('#input');
    await page.keyboard.sendCharacter('hello world');
    await page.click('#reset');

    expect(await page.evaluate('__reset_event__')).toEqual(true);

    await page.click('#submit');
    const submitEventDetail = await page.evaluate('window.__submit_event_detail__');
    expect(submitEventDetail).toMatchSnapshot();
  });
});

