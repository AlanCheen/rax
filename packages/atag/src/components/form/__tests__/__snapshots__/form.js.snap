// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form reset 1`] = `
Object {
  "value": Object {
    "checkbox": Array [
      "checkbox1",
    ],
    "input": "",
    "radio": "",
    "slider": 50,
    "switch": false,
  },
}
`;

exports[`form submit 1`] = `
Object {
  "value": Object {
    "checkbox": Array [
      "checkbox1",
      "checkbox2",
    ],
    "input": "hello world",
    "radio": "radio1",
    "slider": 50,
    "switch": false,
  },
}
`;
